### ISSUE TITLE
Rancher node agent docker shim hangs and agent become zombie process 

### ISSUE THREAD

**Issue by slououou** [rancher/rancher#24307]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Not known yet

**Result:**
Rancher node agent become zombie process.

**Other details that may be helpful:**
```
91995 ?        Sl     4:11 docker-containerd-shim -namespace moby -workdir /var/lib/docker/containerd/daemon/io.containerd.runtime.
92015 ?        Zsl   18:43  \_ [agent] <defunct>
  6593 ?        Z      0:00  \_ [su] <defunct>
121851 ?        Z      0:00  \_ [su] <defunct>
122325 ?        Z      0:00  \_ [su] <defunct>
122558 ?        Z      0:00  \_ [su] <defunct>
122805 ?        Z      0:00  \_ [su] <defunct>
123048 ?        Z      0:00  \_ [su] <defunct>
123154 ?        Z      0:00  \_ [su] <defunct>
123622 ?        Z      0:00  \_ [su] <defunct>
123837 ?        Z      0:00  \_ [su] <defunct>
123933 ?        Z      0:00  \_ [su] <defunct>
124093 ?        Z      0:00  \_ [su] <defunct>
124320 ?        Z      0:00  \_ [su] <defunct>
124465 ?        Z      0:00  \_ [su] <defunct>
124573 ?        Z      0:00  \_ [su] <defunct>
124871 ?        Z      0:00  \_ [su] <defunct>
125019 ?        Z      0:00  \_ [su] <defunct>
125165 ?        Z      0:00  \_ [su] <defunct>
125256 ?        Z      0:00  \_ [su] <defunct>
125412 ?        Z      0:00  \_ [su] <defunct>
125544 ?        Z      0:00  \_ [unshare] <defunct>
125708 ?        Z      0:00  \_ [unshare] <defunct>
126224 ?        Z      0:00  \_ [su] <defunct>
127399 ?        Z      0:00  \_ [su] <defunct>
127486 ?        Z      0:00  \_ [su] <defunct>
127792 ?        Z      0:00  \_ [su] <defunct>
127894 ?        Z      0:00  \_ [su] <defunct>
128092 ?        Z      0:00  \_ [unshare] <defunct>
128242 ?        Z      0:00  \_ [su] <defunct>
128334 ?        Z      0:00  \_ [su] <defunct>
128824 ?        Z      0:00  \_ [su] <defunct>
129083 ?        Z      0:00  \_ [su] <defunct>
129231 ?        Z      0:00  \_ [su] <defunct>
129449 ?        Z      0:00  \_ [unshare] <defunct>
129450 ?        Z      0:00  \_ [su] <defunct>
129704 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
129711 ?        Z      0:00  \_ [unshare] <defunct>
129712 ?        Z      0:00  \_ [su] <defunct>
129940 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
129946 ?        Z      0:00  \_ [unshare] <defunct>
129947 ?        Z      0:00  \_ [su] <defunct>
12224 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
12231 ?        Z      0:00  \_ [unshare] <defunct>
14425 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
14434 ?        Z      0:00  \_ [unshare] <defunct>
14437 ?        Z      0:00  \_ [su] <defunct>
101280 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
101287 ?        Z      0:00  \_ [unshare] <defunct>
101288 ?        Z      0:00  \_ [su] <defunct>
109861 ?        Z      0:00  \_ [unshare] <defunct>
109862 ?        Z      0:00  \_ [su] <defunct>
20414 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
20427 ?        Z      0:00  \_ [unshare] <defunct>
20429 ?        Z      0:00  \_ [su] <defunct>
24147 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
24155 ?        Z      0:00  \_ [su] <defunct>
39540 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
39549 ?        Z      0:00  \_ [unshare] <defunct>
40916 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
40923 ?        Z      0:00  \_ [unshare] <defunct>
40924 ?        Z      0:00  \_ [su] <defunct>
73377 ?        Z      0:00  \_ [unshare] <defunct>
73379 ?        Z      0:00  \_ [su] <defunct>
26166 ?        Zs     0:00  \_ [kubectl-shell.s] <defunct>
26182 ?        Z      0:00  \_ [unshare] <defunct>
28357 ?        Z      0:00  \_ [docker-runc] <defunct>
28371 ?        Z      0:00  \_ [docker-containe] <defunct>
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
hahn025 | v2.2.9
User Interface | v2.2.98
Helm | v2.10.0-rancher11
Machine | v0.15.0-rancher8-1
- Installation option (single install/HA):single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Imported
- Machine type (cloud/VM/metal) and specifications (CPU/memory): 
VM
"capacity": {
"cpu": "4",
"ephemeral-storage": "32485892Ki",
"hugepages-1Gi": "0",
"hugepages-2Mi": "0",
"memory": "7990260Ki",
"pods": "64"
},
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.8", GitCommit:"211047e9a1922595eaa3a1127ed365e9299a6c23", GitTreeState:"clean", BuildDate:"2019-10-15T12:11:03Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-20T11:40:58Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:           18.06.2-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        6d37f41
 Built:             Sun Feb 10 03:46:03 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.2-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       6d37f41
  Built:            Sun Feb 10 03:48:29 2019
  OS/Arch:          linux/amd64
  Experimental:     true
```

**Comment by slououou** [comment]
We've found that the session opened in rancher "launch kubectl" will tirger four process under agent:
\kubectl-shll.s
  \unshare
    \su
      \bash
when the console was closed (by click "close" button), those processes still exist. Also we've found that "lauch kubectl" always connected to the same agent node(or always the same agent node connect back). This will cause high mem usage for agent node process that may lead to a oom in kernel. Once the agent is killed, all those processes become zombie, including the agent, then docker shim process hangs.
